spec.html

```diff
@@ -25893,8 +25956,148 @@ <h1>
+        <emu-table id="table-loadedmodulerequest-fields" caption="Additional fields of LoadedModuleRequest Records">
+          <table>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value Type
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[Module]]
+              </td>
+              <td>
+                a Module Record
+              </td>
+              <td>
+                The loaded module corresponding to this module request
+              </td>
+            </tr>
+          </table>
+        </emu-table>
+
+        <p>An <dfn id="importattribute-record" variants="ImportAttribute Records">ImportAttribute Record</dfn> consists of the following fields:</p>
+        <emu-table id="table-importattribute-fields" caption="ImportAttribute Record fields">
+          <table>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value Type
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[Key]]
+              </td>
+              <td>
+                String
+              </td>
+              <td>
+                The attribute key
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[Value]]
+              </td>
+              <td>
+                String
+              </td>
+              <td>
+                The attribute value
+              </td>
+            </tr>
+          </table>
+        </emu-table>
+
+        <emu-clause id="sec-ModuleRequestsEqual" type="abstract operation">
+          <h1>
+            ModuleRequestsEqual (
+              _left_: a ModuleRequest Record or a LoadedModuleRequest Record,
+              _right_: a ModuleRequest Record or a LoadedModuleRequest Record,
+            ): a Boolean
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd></dd>
+          </dl>
+
+          <emu-alg>
+            1. If _left_.[[Specifier]] is not _right_.[[Specifier]], return *false*.
+            1. Let _leftAttrs_ be _left_.[[Attributes]].
+            1. Let _rightAttrs_ be _right_.[[Attributes]].
+            1. If the number of elements in _leftAttrs_ is not the number of elements in _rightAttrs_, return *false*.
+            1. For each ImportAttribute Record _l_ of _leftAttrs_, do
+              1. Let _found_ be *false*.
+              1. For each ImportAttribute Record _r_ of _rightAttrs_, do
+                1. If _l_.[[Key]] is _r_.[[Key]], then
+                  1. If _l_.[[Value]] is _r_.[[Value]], then
+                    1. Set _found_ to *true*.
```

Comment:
```suggestion
                    1. Assert: _found_ is *false*.
                    1. Set _found_ to *true*.
```

Right? Because the [[Key]] fields come from static `import` syntax or from a call to EnumerableOwnProperties.

Reply:
Yes correct, there can only be one matching key-value pair.